sklearn/linear_model/_bayes.py

```diff
@@ -226,8 +244,31 @@ def fit(self, X, y, sample_weight=None):
         self : object
             Returns the instance itself.
         """
+
         self._validate_params()
 
+        max_iter = self.max_iter
+        # TODO(1.4) Remove
+        if max_iter is None:
+            if self.n_iter != "deprecated":
+                warnings.warn(
+                    "'n_iter' was renamed to 'max_iter' in version 1.2 and "
+                    "will be removed in 1.4",
+                    FutureWarning,
+                )
+                max_iter = self.n_iter
+            else:
+                max_iter = 300
+        else:
+            # Still generate a warning when n_iter is used with max_iter
+            # n_iter is ignored and max_iter is used
+            if self.n_iter != "deprecated":
+                warnings.warn(
+                    "'n_iter' was renamed to 'max_iter' in version 1.2 and "
+                    "will be removed in 1.4",
+                    FutureWarning,
+                )
```

Comment:
This makes the code flow a bit simpler

```suggestion
        # TODO(1.4) Remove
        if self.n_iter != "deprecated":
            warnings.warn(
                "'n_iter' was renamed to 'max_iter' in version 1.2 and "
                "will be removed in 1.4",
                FutureWarning,
            )
            max_iter = self.n_iter
        elif max_iter is None:
            max_iter = 300
```

Reply:
Thanks @Micky774, I had initially done it this way to only update `max_iter` if `max_iter` is `None` and `self.n_iter != 'deprecated'`
I thought it would be consistent ito favour `max_iter` over `n_iter` and we ignore `n_iter` when `max_iter` is set. (Still generating a warning)
I think the change you have suggested will use the user's `n_iter` over `max_iter` if both of them are set.

Reply:
I've updated the suggestion. Generally, if both are set I prefer to throw a `ValueError` and require the user to explicitly choose one or the other, rather than letting it off with a warning. What do you think?

Reply:
This is a better approach. I agree with your suggestion. I will make the necessary changes in both the code and tests. Thanks

Reply:
Indeed, we don't want to allow setting both parameters at the same time.